src/main/java/sirius/db/mixing/query/constraints/FilterFactory.java

```diff
@@ -376,7 +376,7 @@ public C not(@Nullable C constraint) {
      * @param constraint the constraint to invert
      * @return the inverted constraint
      */
-    @Nonnull
+    @Nullable
```

Comment:
**Question:** Are all these nullability changes correct? How can a constraint inversion of a nonnull constraint return a nullable constraint?

Reply:
e.g. an And constraint with zero clauses might be optimized down to "null" and then inverted to "null" - we might also mark the parameter as nullable aswell

Reply:
But isn't the parameter also nullable then?